File: examples/sync/bitmart_public.py
import dcex


def main() -> None:
    client = dcex.bitmart()

    pairs = client.get_trading_pairs()
    print(pairs)

    depth = client.get_depth(product_symbol="BTC-USDT-SWAP")
    print(depth)


if __name__ == "__main__":
    main()

File: crates/dcex/examples/bitmart_public.rs
use std::time::Duration;

use dcex::exchanges::bitmart::BitmartClient;

#[tokio::main]
async fn main() -> dcex::Result<()> {
    let client = BitmartClient::public(Duration::from_secs(10))?;
    let response = client.get_ticker_of_a_pair("BTC-USDT-SPOT").await?;
    println!("{}", response.data);
    Ok(())
}
